worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/WorldEditPlugin.java

```diff
@@ -246,10 +245,11 @@ private void initializeRegistries() {
         }
         // Entity
         for (org.bukkit.entity.EntityType entityType : org.bukkit.entity.EntityType.values()) {
-            String mcid = entityType.getName();
-            if (mcid != null) {
-                String lowerCaseMcId = mcid.toLowerCase(Locale.ROOT);
-                EntityType.REGISTRY.register("minecraft:" + lowerCaseMcId, new EntityType("minecraft:" + lowerCaseMcId));
+            try {
+                String entityTypeKey = entityType.getKey().toString();
+                EntityType.REGISTRY.register(entityTypeKey, new EntityType(entityTypeKey));
+            } catch (IllegalArgumentException e) {
+                // EntityType.UNKNOWN has no key
```

Comment:
For the two instances of this where the enum is iterated, could it instead just check that it's not the UNKNOWN value rather than let it throw an exception? Exceptions are vastly slower than doing a comparison a few times, and look a lot messier. Blindly catching an exception here also might cause us to miss actual errors in the future too